starwars_api/models.py

```diff
@@ -55,27 +59,56 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        page_data = self._get_page_data()
+        self.total_records = page_data['count']
+        self.records = page_data['results']
+
+        self.collected = 0
+        self.counter = 0
+        self.page = 1  # page index starts at 1
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
+
+        while True:
+            if self.collected == self.total_records:
+                raise StopIteration
+            if self.counter > len(self.records) - 1:
+                # get next page records and reset counter
+                self.page += 1
+                page_data = self._get_page_data(page_number=self.page)
+
+                self.records = page_data['results']
+                self.counter = 0
+
+            elem = eval((self.RESOURCE_NAME).capitalize())(
+                self.records[self.counter])
+
+            self.counter += 1
+            self.collected += 1
+            return elem
```

Comment:
Very clean paging solution! I like how you keep track of 'collected' items and therefore don't need to check if next page exists!